Ghidra/Processors/V850E2M/data/languages/Helpers/Macros.sinc

```diff
@@ -0,0 +1,249 @@
+##################################################
+#####				Macros					 #####
+##################################################
+
+
+##### CARRY-Flag #####
+
+macro set_CY_pos(var1, var2)
+{
+	$(CY) = carry(var1, var2);
+}
+
+macro set_CY_pos2(var1, var2, var3)
+{
+	local var12 = var1 + var2;
+	$(CY) = carry(var1, var2) || carry(var12, var3);
+}
+
+macro set_CY_neg(var1, var2)
+{
+	$(CY) = var1 < var2;
+}
+
+macro set_CY_neg2(var1, var2, var3)
+{
+	local var23 = var2 + var3;
+	$(CY) = (var1 < var23);
+}
+
+
+
+##### Overflow-Flag #####
+
+macro set_OV_pos(var1, var2)
+{
+	$(OV) = scarry(var1, var2);
+}
+
+macro set_OV_pos2(var1, var2, var3)
+{
+	local var12 = var1 + var2;
+	$(OV) = scarry(var1, var2) || scarry(var12, var3);
+}
+
+macro set_OV_neg(var1, var2)
+{	
+	local A:4 = var1;
+	local B:4 = var2;
+	local R = A - B;
+
+	local A1 = A[31,1];
+	local B1 = B[31,1];
+	local R1 = R[31,1];
+
+	$(OV) = (A1 != B1) && (B1 == R1);
+
+	#OV = 1 if:
+	#pos - neg = neg
+	#neg - pos = pos
+}
+
+macro set_OV_neg2(var1, var2, var3)
+{
+	local A:4 = var1;
+	local B:4 = var2;
+	local C:4 = var3;
+	local R = A - B - C;
+
+	local A1 = A[31,1];
+	local B1 = B[31,1];
+	local R1 = R[31,1];
+
+	$(OV) = (A1 != B1) && (B1 == R1);
+}
+
+
+
+##### S/Z-Flags #####
+
+macro set_S(flag)
+{
+	$(S) = flag s< 0;
+}
+
+macro set_Z(var)
+{
+	$(Z) = var == 0;
+}
+
+
+
+##### General-Flag-Macros #####
+
+macro set_general_flags_pos(var1, var2)
+{
+	local res = var1 + var2;
+	set_CY_pos(var1, var2);
+	set_OV_pos(var1, var2);
+	set_S(res);
+	set_Z(res);
+}
+
+macro set_general_flags_neg(var1, var2)
+{
+	local res = var1 - var2;
+	set_CY_neg(var1, var2);
+	set_OV_neg(var1, var2);
+	set_S(res);
+	set_Z(res);
+}
+
+macro set_OV0_S_Z(var)
+{
+	$(OV) = 0;
+	set_S(var);
+	set_Z(var);
+}
+
+
+
+##### General-Macros #####
+
+macro either_or(res, cond, true, false) 
+{
+	res = (true * zext(cond != 0)) + (false * zext(cond == 0));
```

Comment:
it looks like some float functions pass through here, you would need another set for `f+`

Reply:
Ahh alright, if I multiply a float number with an integer, it gets an integer afterwords with the Symbol '*'.  I still had the old view from C

Reply:
I think, its better to change to variant with `if (...) goto <...>` conditions. 
It much easer for a man to understand.